File: src/main/java/toast/pine/Events/PlayerSendFriendInviteEvent.java
package toast.pine.Events;

import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;
import toast.pine.SocialSystem.PlayerSocial;

public class PlayerSendFriendInviteEvent extends Event {

    private static final HandlerList HANDLERS = new HandlerList();

    private final PlayerSocial sender;
    private final PlayerSocial receiver;

    public PlayerSendFriendInviteEvent(PlayerSocial sender, PlayerSocial receiver) {
        this.sender = sender;
        this.receiver = receiver;
    }

    public PlayerSocial getSender() {
        return this.sender;
    }

    public PlayerSocial getReceiver() {
        return this.receiver;
    }

    @Override
    public @NotNull HandlerList getHandlers() {
        return HANDLERS;
    }
}

File: src/main/java/pine/toast/toastrpg/events/PlayerSendFriendInviteEvent.kt
package pine.toast.toastrpg.events

import org.bukkit.event.Event
import org.bukkit.event.HandlerList
import pine.toast.toastrpg.socialsystem.PlayerSocial

class PlayerSendFriendInviteEvent(private val sender: PlayerSocial, private val receiver: PlayerSocial) : Event() {

    private val handlers = HandlerList()
    fun getSender(): PlayerSocial {
        return sender
    }

    fun getReceiver(): PlayerSocial {
        return receiver
    }

    override fun getHandlers(): HandlerList {
        return handlers
    }


}
